distributed/http/worker/prometheus/core.py

```diff
@@ -161,6 +146,128 @@ def collect(self):
+          unpickle = spill_time_total.unpickle   / spill_count_total.disk_read
+          read     = spill_time_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per key:
+          write    = spill_bytes_total.disk_write / spill_count_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per second:
+          write    = spill_bytes_total.disk_write / spill_time_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_time_total.disk_read
+        """
+        try:
+            get_metrics = self.server.data.get_metrics  # type: ignore
+        except AttributeError:
+            return  # spilling is disabled
+        metrics = get_metrics()
+
+        total_bytes = CounterMetricFamily(
+            self.build_name("spill_bytes_total"),
+            "Total size of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_bytes.add_metric([k], metrics[f"{k}_bytes_total"])
+
+        total_counts = CounterMetricFamily(
+            self.build_name("spill_count_total"),
+            "Total number of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_counts.add_metric([k], metrics[f"{k}_count_total"])
+
+        total_times = CounterMetricFamily(
+            self.build_name("spill_time_total"),
+            "Total time spent spilling/unspilling since the latest worker restart",
+            labels=["event"],
+        )
+        for k in ("pickle", "disk_write", "disk_read", "unpickle"):
+            total_times.add_metric([k], metrics[f"{k}_time_total"])
+        yield total_times
+
+        max_times = GaugeMetricFamily(
+            self.build_name("spill_time_per_key_max"),
+            "Maximum time spent spilling/unspilling a single key "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        for k in ("pickle", "disk_write", "disk_read", "unpickle"):
+            max_times.add_metric([k], metrics[f"{k}_time_per_key_max"])
+        yield max_times
+
+        max_counts = GaugeMetricFamily(
+            self.build_name("memory_count_max"),
+            "Maximum number of keys in memory since the latest worker restart",
+            labels=["where"],
+        )
+        max_counts.add_metric(["memory"], metrics["memory_count_max"])
+        max_counts.add_metric(["disk"], metrics["disk_count_max"])
+        max_counts.add_metric(["total"], metrics["count_max"])
+        yield max_counts
```

Comment:
@ntabris may I have your opinion here?

- I'm exposing max measures. The idea is to catch flares that are faster than 5 seconds (or whatever the prometheus polling interval is).
- memory, disk, and total are not additive (and `max(memory + disk)` != `max(memory) + max(disk)`); however I think they should be grouped together somehow. 

What is the best practice here? Is `gauge` the correct metric type?

Reply:
My impression is that exposing maxes to prometheus is a little iffy but I understand why we're going it and don't have better suggestion at present, so this seems fine.

Just to check my understanding, **at any specific time** `toatl = memory + disk`, but because these are maxes **over time**, `max(total) != max(memory) + max(disk)`, right?

Reply:
correct